### PR TITLE
Restrict products

### PR DESCRIPTION
**Pull request by aparnakr**
# Task: Restrict products 

## High Level Description
Only allow creating options products which are fully collateralized which won't require any liquidation mechanism. 

Specific Changes
Added a require statement in whitelist to check that only put option products where the strike == collateral are allowed and call options where underlying == collateral are allowed. 

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [ ] Is your code up to date with the spec? 
- [ ] Have you added your tests to the testing doc?

### CODE REVIEW

**Comment by antoncoding** [comment]
Plz add tests for this line!

**Comment by haythem96** [comment]
This will break the integration tests in `Yield Farming: Naked Put Option closed before expiry flow`, do u want to just ignore those for now by adding `xit`

File: contracts/Whitelist.sol [hunk 1]
```diff
@@ -167,7 +171,7 @@ contract Whitelist is Ownable {
 
     /**
      * @notice allows the owner to whitelist a collateral address
-     * @dev can only be called from the owner address
+     * @dev can only be called from the owner address. Don't whitelist otokens as collateral in the current version.
```
Comment (antoncoding) [inline contracts/Whitelist.sol:174]: Why do we want to add this description? otokens are never meant to be a collateral in our system right?

**Review by antoncoding — APPROVED** [review]

File: contracts/Whitelist.sol (hunk 1 above)
Comment (haythem96) [reply contracts/Whitelist.sol:174]: Yeah I think better to update this comment. Otoken are already collateral in our system, but they are used in the long Otoken array, so maybe make sense to just explain that this functions is meant for collateral assets that are not an Otoken?

**Review by haythem96 — APPROVED** [review]

**Pull request merged**